### DISCUSSION TITLE
WiFi Mode Capability

### DISCUSSION THREAD

**Discussion by bwjohns4** [arendst/Tasmota#14552] [Q&A] [answered]
I have a question regarding the WiFi modes available. Tasmota documentation implies you can put into b/g/n mode, b/g mode, or just b mode while the underlying ESP8266 enum (below) implies that no combinations of modes are available just N or G or B, rather than B/G/N OR B/G OR B mode. Would someone be able to provide clarity on this?
```
typedef enum {
    PHY_MODE_11B    = 1,
    PHY_MODE_11G    = 2,
    PHY_MODE_11N    = 3
} phy_mode_t;
```

**Comment by ascillato** [comment, marked as answer]
Hi,

The Espressif SDK is the one that does the combination. You can not select any other mode for ESP8266 rather than the ones provided in the Wi-Fi closed source routines of the SDK:

* IEEE 802.11b/g/n
* IEEE 802.11b/g
* IEEE 802.11b

This means that the device will support that, but the one that defines which mode is going to be used is the Wi-Fi router. With the `Wifi` Command you only define which ones will be the capabilities of the device. The default and the recommended one is IEEE 802.11b/g/n

That ENUM is explained in the SDK documentation from Espressif, stating that behaves as above. Also a lot of testing has been done that validates that.

Note: It is easy to test for yourself if you want. Put your device in B/G/N and set your router to be 11n-only, then 11g and then 11b and your Tasmota will connect in all those modes. Then try B/G in Tasmota and in your router the 3 options, and Tasmota will connect only in B and G. And also for B only mode.

More info at: https://github.com/arendst/Tasmota/pull/12292
